### PR TITLE
[4.0] Correct Table::hasField() uses

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request related to Issue #21445 .

### Summary of Changes

Don't put fields through `getColumnAlias()` before passing them to `hasField()`. It already does that.

### Testing Instructions

Code review.

CC @wilsonge @alikon.

### CODE REVIEW

File: libraries/src/MVC/Model/FormModel.php [hunk 1]
```diff
@@ -98,17 +98,17 @@ public function checkin($pk = null)
...
 			{
 				return true;
 			}
 
+			$checkedOutField = $this->getColumnAlias('checked_out');
+			$checkedOutTimeField = $this->getColumnAlias('checked_out_time');
+
 			// Check if this is the user having previously checked out the row.
-			if ($table->{$checkedOutField} > 0 && $table->{$checkedOutField} != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
+			if ($table->$checkedOutField > 0 && $table->$checkedOutTimeField != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
```
Comment (csthomas) [inline libraries/src/MVC/Model/FormModel.php]:
```suggestion
			if ($table->$checkedOutField > 0 && $table->$checkedOutField != $user->get('id') && !$user->authorise('core.admin', 'com_checkin'))
```

Do not compare time with user_id.

File: libraries/src/MVC/Model/FormModel.php [hunk 2]
```diff
@@ -151,19 +151,16 @@ public function checkout($pk = null)
...
+			if (!$table->hasField('checked_out') || !$table->hasField('checked_out_time'))
 			{
 				return true;
 			}
 
 			$user = Factory::getUser();
 
 			// Check if this is the user having previously checked out the row.
-			if ($table->{$checkedOutField} > 0 && $table->{$checkedOutField} != $user->get('id'))
+			if ($table->{$table->getColumnAlias('checked_out')} > 0 && $table->{$table->getColumnAlias('checked_out_time')} != $user->get('id'))
```
Comment (csthomas) [inline libraries/src/MVC/Model/FormModel.php]: The same

Comment (SharkyKZ) [reply libraries/src/MVC/Model/FormModel.php]: Fixed. Thanks.

**Review by wilsonge — APPROVED** [review]

File: libraries/src/Table/Table.php [hunk 3]
```diff
@@ -1282,6 +1280,10 @@ public function checkIn($pk = null)
 				throw new \UnexpectedValueException('Null primary key not allowed.');
 			}
 		}
+
```
Comment (Quy) [inline libraries/src/Table/Table.php]:
```suggestion

```

File: libraries/src/Table/Table.php [hunk 4]
```diff
@@ -1282,6 +1280,9 @@ public function checkIn($pk = null)
 				throw new \UnexpectedValueException('Null primary key not allowed.');
 			}
 		}
+		// Get column names.
```
Comment (Quy) [inline libraries/src/Table/Table.php:1284]:
```suggestion

		// Get column names.
```

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
Thanks!

**Comment by infograf768** [comment]
@SharkyKZ @wilsonge 
I have an error after this patch when close or save and close a plugin
`Call to undefined method Joomla\Component\Plugins\Administrator\Model\PluginModel::getColumnAlias() `

I think (as I tested here) that
`$checkedOutField = $this->getColumnAlias('checked_out');`
is wrong
and should be changed to
`$checkedOutField = $table->getColumnAlias('checked_out');`

in libraries/src/MVC/Model/FormModel.php line 107

**Comment by SharkyKZ** [comment]
@infograf768 Thanks, it's a typo. Test #23220 please.

**Pull request merged**